_posts/en/newsletters/2020-08-12-newsletter.md

```diff
@@ -0,0 +1,206 @@
+
+    One problem with that mechanism is that both an attacker and an
+    honest user could both try spending from the same previous state.
+    Miners and relay nodes would only keep one of those transactions in
+    their mempool, and there may be ways the attacker could ensure it
+    was an old state.  Another problem is that the attacker could
+    possibly trick the honest user into spending from a state that relay
+    nodes hadn't seen and so relay nodes would possibly reject the
+    unconfirmed transaction.  Although neither of these problems would
+    prevent the ultimate state from eventually being confirmed onchain,
+    they could be used for [transaction pinning][topic transaction
+    pinning] to prevent one of the honest user's time-sensitive
+    transactions from confirming in time.  These problems are similar to
+    the attack against LN payment atomicity described in [Newsletter
+    #95][news95 ln atomicity].  One proposed mitigation would be to have
+    special nodes (perhaps part of LN routing software) that looked for
+    cases where eltoo was being used and could use their knowledge of
+    that protocol to tell miners which transaction would truly be most
+    profitable to mine.
+
+## Field report: How segwit and batching could have saved half a billion dollars in fees
+
+{% include articles/veriphi-segwit-batching.md %}
+
+## Bitcoin Core PR Review Club
+
+*In this monthly section, we summarize a recent [Bitcoin Core PR Review Club][]
+meeting, highlighting some of the important questions and answers.  Click on a
+question below to see a summary of the answer from the meeting.*
+
+[Implement ADDRv2 support (part of BIP155)][review club #19031] is a PR
+([#19031][Bitcoin Core #19031]) by Vasil Dimov that proposes a partial
+implementation of the [BIP155][] `addrv2` P2P message.
+
+The discussion covered the current `addr` message in Bitcoin Core, the BIP155
+`addrv2` message, how to signal support for `addrv2` on the P2P network, and the
+necessity of `addrv2` for the Bitcoin network to upgrade to Tor v3. Tor v2
+deprecation begins mid-September!
+
+{% include functions/details-list.md
+  q0="How does Bitcoin Core handle network addresses?"
+  a0="Bitcoin Core currently sees all peer addresses as being more or less in a
+      [single][19031 single] IPv6-like space. IPv4 and Tor v2 addresses are
+      serialized into [16 bytes][19031 fake] encoded as “fake” IPv6 addresses
+      from specific IPv6 networks reserved for them. For instance, Tor v2
+      addresses are persisted using the [“onioncat” IPv6 range][19031 onioncat].
+      The 16-byte maximum length of IPv6 increasingly limits which endpoints
+      Bitcoin nodes can connect to."
+  a0link="https://bitcoincore.reviews/19031#l-35"
+
+  q1="How are peer addresses stored?"
+  a1="Addresses are saved in the `peers.dat` file in a 16-byte format, which
+      corresponds to the length of IPv6 and Tor v2 addresses; the smaller-sized
```

Comment:
`s/ and Tor v2//` (TORv2 addresses are 10 bytes)

Reply:
The review club study notes might need to be corrected: "Relevant to PR 19031 is the fact that Tor v2 addresses are 16 characters long, nicely fitting in the 16 bytes needed for an IPv6 address."

Reply:
(Yes, there is the 6-byte onion prefix of `0xFD,0x87,0xD8,0x7E,0xEB,0x43`).

Reply:
Should this instead be `s/and Tor v2/and onioncat-encoded Tor v2/` ?

Reply:
Yes, "onioncat-encoded Tor v2" would be correct. Since we also store IPv4 in a similar way, maybe reword to: "which corresponds to the length of IPv6 and other addresses encoded as IPv6".

Reply:
Within the context of this paragraph and the preceding ones, maybe:

"which corresponds to the length of IPv6 addresses; the smaller-sized IPv4 and Tor v2 addresses are therefore padded to fit."

Reply:
(was tempted to link to `CNetAddr::SetLegacyIPv6`, but this code has changed considerably from 0.20 to master and the stable release code isn't as clear about it as it is now).

Reply:
> The review club study notes might need to be corrected

done (for info)